docs/pages/Cardinal/Cross_Shard_Communication/evm_to_cardinal.mdx

```diff
@@ -0,0 +1,188 @@
+import { Callout } from 'nextra/components'
+
+<Callout type={"warning"}>
+    EVM to Cardinal communication is an experimental feature. The API is not final and bugs are expected to be encountered.
+</Callout>
+
+
+# EVM To Cardinal Communication
+
+The EVM base shard provides a precompile, called Router, that enables smart contracts to read data from and send messages to Cardinal game shards.
+
+For messages, router operates asynchronously, which requires a callback mechanism to be used. Messages are only forwarded from Router's `sendMessage` method once the EVM transaction that called it successfully executes. Shortly after, the result will be available for consumption via Router's `messageResult` method.
+
+Queries are synchronous and do not require a callback.
+
+## Precompile Address
+
+`0x356833c4666fFB6bFccbF8D600fa7282290dE073`
+
+## Precompile Interface
+
+```solidity
+interface IRouter {
+    function sendMessage(bytes memory message, string memory messageID, string memory namespace) external returns (bool);
+
+    function messageResult(string memory txHash) external returns (bytes memory, string memory, uint32);
+
+    function query(bytes memory request, string memory resource, string memory namespace)
```

Comment:
newlining/indenting the external and returns portions of these signatures looks clean. I'd suggest the same treatment for the sendMessage and messageResults lines.